CHANGELOG.md

```diff
@@ -19,12 +20,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added optional props to specify icon titles in the `AudioInputControl` and `ContentShareControl` components.
 - Added optional props to specify the dropdown text that shows when no video quality is selected, in the `QualitySelection` component.
 - Added observer to `selectAudioInputDeviceError` and `selectVideoInputDeviceError` to deliver the error from SDK level to client application level.
-  
+- Added optional parameter `isDevicePreviewSkipped` in `meetingManager.join()` to suppress device permission request.
```

Comment:
Hi Richard, thank you so much for giving me these comments. I really had a hard time naming them. Your suggestion is right what I need.
The main goal of this PR is to provide our customers with a simple way to achieve the view-only mode, that is, users join a meeting without audio input device and video input device, and no device permission request appears during the joining process.
At the beginning this optional parameter is called `joinAsSpectator`, and we find it too specific, then we change the name to 'skipDevicePreview` to make it more generic.
Do you have any better idea for its naming?

Reply:
The `enum DeviceLables` is really a great idea. We could use it to have more flexible controls over the device permission. I will make the improvement!

For second issue, I add a checkbox in `MeetingForm` in meeting demo. So user could select between normal mode and view-only mode when they join a meeting. 

<img width="482" alt="image" src="https://user-images.githubusercontent.com/20471438/117872344-6a470100-b253-11eb-80a9-70dbb4c7c302.png">

I made the `MeetingControls` to be rendered conditionally according to the `devicePermissionStatus` exposed by the `useDevicePermissionStatus`.

For normal mode, everything works as before.

For view-only mode, user skips the device setting page, and join the meeting with `null` devices, the speaker works as usual. The `MeetingControls` displays only `EndMeetingControl` and `DevicePermissionControl`. 

<img width="138" alt="image" src="https://user-images.githubusercontent.com/20471438/117873640-e261f680-b254-11eb-87e9-c8a7ab5ec7dc.png">


'DevicePermissionControl` can trigger `audioVideo.listXxxxxXxxxxDevices()` function, user could grant device permission through this component. 

<img width="1146" alt="image" src="https://user-images.githubusercontent.com/20471438/117872940-1b4d9b80-b254-11eb-844e-a9eb69767b4e.png">

After the permission is granted, the `MeetingControls` is rendered like normal way. User could select and set the device through Device Controls in `MeetingControls`. And at this time everything works like the normal mode.

<img width="528" alt="image" src="https://user-images.githubusercontent.com/20471438/117873019-302a2f00-b254-11eb-880a-ea701af8f0f7.png">

I think it would be more practical to do it like this, device permission is delayed instead of suppressed.

Reply:
It's not really "Skip device preview", it's "join without audio or video", no?

Be careful about the behavior when the user chooses Deny, or doesn't save permissions — test in Firefox rather than Chrome.

Reply:
You're right. I renamed the checkbox label.

For the behavior here, there are two issues right now. 
* The first one is https://github.com/aws/amazon-chime-sdk-js/issues/474. I gonna work on it after this PR being merged.

* The second is that we use the state of `useDevicePermissionStatus` hook to render the `MeetingControls` conditionally. But the it cannot be synchronized with the actual device permission status, if the browser saves the device permission. e.g. User gave "Always Allow" device permission to browser when using the app last time. This time user opens the Application, the actual permission status is "Granted", but the state returned by the hook is "Unset". Since we gonna add controls to device permission, it should be much better to solve this issue. But currently I can't find some good way to detect the device permission at the very beginning. 

Any suggestion would be helpful!

Reply:
This is why I suggest not trying. Build two experiences, handle failures, and to transition between the two, invoke the device label function. Do not try to represent permission status in React state; that hook probably should not exist.

Reply:
To be clear: you _cannot_ find out whether the user has granted permissions without enumerating devices and checking to see whether they have labels.